File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Protocol/ProtocolDraft.vb
Imports VecompSoftware.DocSuiteWeb.Data

<Serializable()>
Public Class ProtocolDraft
    Inherits AuditableDomainObject(Of Guid)

#Region " Fields "
    Private _protocol As Protocol
#End Region

#Region " Properties "
    Public Overridable Property Incremental As Integer

    Public Overridable Property Year As Short?

    Public Overridable Property Number As Integer?

    Public Overridable Property IsActive As Short

    Public Overridable Property Description As String

    Public Overridable Property Data As String

    Public Overridable Property DraftType As Integer

    Public Overridable Property Protocol As Protocol
        Get
            Return _protocol
        End Get
        Set(ByVal value As Protocol)
            _protocol = value
            If value IsNot Nothing Then
                Year = value.Year
                Number = value.Number
            End If
        End Set
    End Property

    Public Overridable Property Collaboration As Collaboration
#End Region

#Region "Constructor"
    Public Sub New()

    End Sub
#End Region

End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Protocols/ProtocolDraft.cs
using System;
using VecompSoftware.DocSuiteWeb.Entity.Collaborations;

namespace VecompSoftware.DocSuiteWeb.Entity.Protocols
{
    public class ProtocolDraft : DSWBaseEntity
    {
        #region [ Constructor ]
        public ProtocolDraft() : this(Guid.NewGuid()) { }

        public ProtocolDraft(Guid uniqueId)
            : base(uniqueId)
        {
        }
        #endregion

        #region [ Properties ]
        public virtual bool IsActive { get; set; }

        public virtual string Description { get; set; }

        public virtual string Data { get; set; }

        public virtual int DraftType { get; set; }
        #endregion

        #region [ Navigation Properties ]
        public virtual Collaboration Collaboration { get; set; }
        public virtual Protocol Protocol { get; set; }
        #endregion
    }
}
